src/asl.ts

```diff
@@ -1499,8 +1499,20 @@ export class ASL {
         // If the identifier references a parameter expression and that parameter expression
         // is in a FunctionDecl and that Function is at the top (no parent).
         // This logic needs to be updated to support destructured inputs: https://github.com/functionless/functionless/issues/68
-        if (ref && isParameterDecl(ref) && isFunctionDecl(ref.parent)) {
+        if (
+          ref &&
+          isParameterDecl(ref) &&
+          isFunctionDecl(ref.parent) &&
```

Comment:
With my change, this is changing - we can no longer rely on the top being a function decl. Need to check for parent is undefined.

Reply:
Wouldn't parent be undefined for any top level function? Wouldn't we'd want `ref.parent === sfn.decl`?

Reply:
`&& ref.parent === this.decl`